README.md

```diff
@@ -274,189 +283,566 @@ This is defined in libocpp/include/ocpp/v16/charge_point.hpp and takes the follo
 
-- on_reservation_end
+- `on_reservation_end`
 
   Notifies libocpp that a reservation has ended
 
 #### Authorization
-In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge)
-To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object
 
-## Integrate this library with your Charging Station Implementation for OCPP2.0.1
-TODO
+In EVerest authorization is handled by the Auth module and various auth token providers and validators. The OCPP module acts as both a token provider (for pre validated tokens in RemoteStartTransactions) and a token validator (using the authorize requests, or plug & charge).
+
+To use libocpp as a auth token validator (e.g. before starting a transaction) you can call the "authorize_id_token" function of the ChargePoint object.
+
+## Usage for OCPP 2.0.1
+
+### Smart Charging
+
+Work to fully support OCPP 2.0.1 Smart Charging is ongoing. Most functional requirements for General Smart Charging use cases (that is, K01–K10) are now supported. For an up-to-date overview of which features are currently supported, please refer to the [OCPP 2.0.1 Status](doc/ocpp_201_status.md) document.
+
+#### K01 SetChargingProfile
+
+Allows the CSMS to influence the charging power or current drawn from a specific EVSE or the
+entire Charging Station over a period of time.
+
+```mermaid
+sequenceDiagram
+    CSMS->>+ChargePoint : SetChargingProfileRequest(call)
+
+    ChargePoint->>+DeviceModel : SmartChargingCtrlrAvailable?
+    DeviceModel-->>-ChargePoint : Component
+
+    rect Red 
+    break SmartChargingCtrlrAvailable = false
+        ChargePoint-->>CSMS : Smart Charging NotSupported CallError
+    end
+    end
+
+    ChargePoint->>+SmartCharging : validate_and_add_profile(call.msg.Profile, call.msg.EVSE ID)
+
+    SmartCharging->>SmartCharging : validate_profile(Profile, EVSE ID)
+
+    rect Red 
+    break Invalid Profile
+        SmartCharging-->>ChargePoint : SetChargingProfileResponse: Rejected
+        ChargePoint-->>CSMS : SetChargingProfileResponse: Rejected
+    end
+    end
+
+    SmartCharging->>+SmartCharging : add_profile(Profile, EVSE ID)
+    SmartCharging->>-EVerest : signal_set_charging_profiles_callback
+
+    SmartCharging-->>-ChargePoint : SetChargingProfileResponse: Accepted
+
+    ChargePoint-->>-CSMS : SetChargingProfileResponse: Accepted
+```
+
+Profile validation returns the following errors to the caller when a Profile
+is `Rejected`:
+
+- `ChargingProfileFirstStartScheduleIsNotZero`
+
+  The `startPeriod` of the first `chargingSchedulePeriod` needs to be 0.
+  [K01.FR.31]
+
+- `ChargingProfileNoChargingSchedulePeriods`
+
+  Happens when the `ChargingProfile` doesn't have any Charging Schedule
+  Periods.
+
```

Comment:
Again, I don't think this should go into the README. I don't even think it should go into the architecture diagram. Instead, it should be a separate document so it can be linked to directly. From a formatting perspective, most error tables are ... tables and not lists.

Reply:
@shankari - just wanted to add a note here too. Do you have a preference on location. If not I was leaning towards the `doc` directory.